### PR TITLE
H2GIS update.

### PR DESCRIPTION
**Pull request by SPalominos**
Make H2GIS compatible with H2database 1.4.199 and the new format of geometries.

### CODE REVIEW

File: h2gis-functions/src/main/java/org/h2gis/functions/factory/H2GISFunctions.java [hunk 1]
```diff
@@ -52,38 +62,111 @@
...
+import org.h2gis.functions.spatial.distance.ST_ClosestCoordinate;
+import org.h2gis.functions.spatial.distance.ST_ClosestPoint;
+import org.h2gis.functions.spatial.distance.ST_FurthestCoordinate;
+import org.h2gis.functions.spatial.distance.ST_LocateAlong;
+import org.h2gis.functions.spatial.distance.ST_LongestLine;
+import org.h2gis.functions.spatial.distance.ST_MaxDistance;
+import org.h2gis.functions.spatial.distance.ST_ProjectPoint;
+import org.h2gis.functions.spatial.distance.ST_ShortestLine;
 import org.h2gis.functions.spatial.earth.ST_GeometryShadow;
-import org.h2gis.functions.spatial.earth.ST_Isovist;
```
Comment (ebocher) [inline h2gis-functions/src/main/java/org/h2gis/functions/factory/H2GISFunctions.java]: Why remove this function ?

File: h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java [hunk 2]
```diff
@@ -61,7 +61,6 @@ public static void feedTableDataFromHeader(DbaseFileHeader header, CreateTableDa
             String fieldsName = header.getFieldName(i);
             final int type = dbfTypeToH2Type(header,i);
             Column column = new Column(fieldsName.toUpperCase(), type);
-            column.setPrecision(header.getFieldLength(i)); // set string length
```
Comment (ebocher) [inline h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java]: The precision of the column cannot be set ?

Comment (SPalominos) [reply h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java]: It has been removed from the Column class in h2database ([here](https://github.com/h2database/h2database/blob/7707e5d42f4967e31c5c7487252adc3cd5564532/h2/src/main/org/h2/table/Column.java#L67))

File: h2gis-functions/src/main/java/org/h2gis/functions/io/file_table/H2Table.java [hunk 3]
```diff
@@ -54,22 +55,29 @@
...
     }
     /**
      * Create row index
      * @param session database session
      */
-    public void init(Session session) {
-        indexes.add(0, new H2TableIndex(driver,this,this.getId()));
+    public void init(Session session) {        
+        IndexColumn indexColumn = new IndexColumn();
+        indexColumn.columnName = "key";
```
Comment (ebocher) [inline h2gis-functions/src/main/java/org/h2gis/functions/io/file_table/H2Table.java]: What about pk as name ?

File: h2gis-functions/src/main/java/org/h2gis/functions/factory/H2GISFunctions.java (hunk 1 above)
Comment (SPalominos) [reply h2gis-functions/src/main/java/org/h2gis/functions/factory/H2GISFunctions.java]: Fixed

File: h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java (hunk 2 above)
Comment (ebocher) [reply h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java]: Maybe something like that
TypeInfo.getTypeInfo(type, precision, scale, null)

Comment (ebocher) [reply h2gis-functions/src/main/java/org/h2gis/functions/io/dbf/DBFEngine.java]: Column column = new Column(fieldsName.toUpperCase(), TypeInfo.getTypeInfo(type, precision, scale, null));

**Pull request merged**